scripts/build-opentracing.sh

```diff
@@ -0,0 +1,114 @@
+git clone -b "$JAEGER_CPP_VERSION" https://github.com/jaegertracing/cpp-client.git /tmp/jaeger-cpp
+git clone -b "${OPENSSL_VERSION}" https://github.com/openssl/openssl.git /tmp/openssl
+
+###################### fix
+export RESTY_PCRE_VERSION="8.41"
+curl -fSL https://ftp.pcre.org/pub/pcre/pcre-${RESTY_PCRE_VERSION}.tar.gz -o /tmp/pcre-${RESTY_PCRE_VERSION}.tar.gz
+cd /tmp/
+tar xzf pcre-${RESTY_PCRE_VERSION}.tar.gz
+
+###################### fix
+# Let's Build opentracing-cpp client
+cd /tmp/opentracing
+mkdir .build
+cd .build
+cmake3 -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF ..
+make -j"$(nproc)"
+make install
+
+# Let's Build jaeger-cpp client
+cd /tmp/jaeger-cpp
+mkdir .build
+cd .build
+cmake3 -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -DJAEGERTRACING_WITH_YAML_CPP=OFF -DBUILD_TESTING=OFF ..
+make -j"$(nproc)"
+make install
+
+#Let's build dynamic modules for opentracing and jaeger, to keep
+#binary compatibility with the openresty installed from the RPM
+#we need to configure it the same way.
+cd /tmp/openresty-"${OPENRESTY_RPM_VERSION}"
+
+./configure -j"$(nproc)" \
+            --with-openssl=/tmp/openssl \
+            --with-pcre=/tmp/pcre-${RESTY_PCRE_VERSION} \
+            --with-file-aio \
+            --with-http_addition_module \
+            --with-http_auth_request_module \
+            --with-http_dav_module \
+            --with-http_flv_module \
+            --with-http_geoip_module=dynamic \
+            --with-http_gunzip_module \
+            --with-http_gzip_static_module \
+            --with-http_image_filter_module=dynamic \
+            --with-http_mp4_module \
+            --with-http_random_index_module \
+            --with-http_realip_module \
+            --with-http_secure_link_module \
+            --with-http_slice_module \
+            --with-http_ssl_module \
+            --with-http_stub_status_module \
+            --with-http_sub_module \
+            --with-http_v2_module \
+            --with-http_xslt_module=dynamic \
+            --with-ipv6 \
+            --with-mail \
+            --with-mail_ssl_module \
+            --with-md5-asm \
+            --with-pcre-jit \
+            --with-sha1-asm \
+            --with-stream \
+            --with-stream_ssl_module \
+            --with-threads \
+            --add-dynamic-module=/tmp/nginx-opentracing/opentracing \
+            --add-dynamic-module=/tmp/nginx-opentracing/jaeger
+
+make -j"$(nproc)"
+
+cd /tmp/openresty-"${OPENRESTY_RPM_VERSION}"/build/nginx-1.13.6/ && make modules
+
+# Move modules to another dir.
+mkdir /opt/app-root/src/nginx-modules/
+mkdir /opt/app-root/src/opentracing-libs/
+cp objs/ngx_http_opentracing_module.so /opt/app-root/src/nginx-modules/
+cp objs/ngx_http_jaeger_module.so /opt/app-root/src/nginx-modules/
+
+# Move libraries
+cp /usr/local/lib/libjaeger* /opt/app-root/src/opentracing-libs/
+cp /usr/local/lib/libopentracing* /opt/app-root/src/opentracing-libs/
+
+# clean
+rm -rf /tmp/*
```

Comment:
This is quite nasty no?

Reply:
not really elegant.. but what do you propose?

Reply:
going to use mktemp